controlpanel/frontend/views/dashboard.py

```diff
@@ -0,0 +1,303 @@
+    permission_required = "api.remove_dashboard_customer"
+
+    def get_redirect_url(self, *args, **kwargs):
+        return reverse_lazy("dashboard-customers", kwargs={"pk": self.kwargs["pk"], "page_no": 1})
+
+    def perform_update(self, **kwargs):
+        dashboard = self.get_object()
+        user_ids = self.request.POST.getlist("customer")
+        try:
+            dashboard.delete_customers_by_id(user_ids)
+        except DeleteCustomerError as e:
+            sentry_sdk.capture_exception(e)
+            messages.error(self.request, f"Failed removing customer{pluralize(user_ids)}")
+        else:
+            messages.success(self.request, f"Successfully removed customer{pluralize(user_ids)}")
+
+
+@method_decorator(feature_flag_required("register_dashboard"), name="dispatch")
+class RemoveDashboardCustomerByEmail(UpdateDashboard):
+    model = Dashboard
+    form = None
+    permission_required = "api.remove_dashboard_customer"
+
+    def get_redirect_url(self, *args, **kwargs):
+        return reverse_lazy("dashboard-customers", kwargs={"pk": self.kwargs["pk"], "page_no": 1})
+
+    def perform_update(self, **kwargs):
+        """
+        Attempts to remove a user from a group, based on their email address
+        """
+        self.form = RemoveCustomerByEmailForm(data=self.request.POST)
+        if not self.form.is_valid():
+            self.request.session["customer_form_errors"] = self.form.errors
+            return messages.error(self.request, "Invalid email address entered")
+
+        dashboard = self.get_object()
+        email = self.form.cleaned_data["email"]
+        try:
+            dashboard.delete_customer_by_email(email)
+        except DeleteCustomerError as e:
+            sentry_sdk.capture_exception(e)
+            return messages.error(
+                self.request, str(e) or f"Couldn't remove customer with email {email}"
+            )
+
+        messages.success(self.request, f"Successfully removed customer {email}")
+
+
+@method_decorator(feature_flag_required("register_dashboard"), name="dispatch")
+class GrantDomainAccess(
+    OIDCLoginRequiredMixin,
+    PermissionRequiredMixin,
+    UpdateView,
+):
+    form_class = GrantDomainAccessForm
+    model = Dashboard
+    permission_required = "api.add_dashboard_domain"
+
+    def get_form_kwargs(self):
+        kwargs = FormMixin.get_form_kwargs(self)
+        if "dashboard" not in kwargs:
+            kwargs["dashboard"] = Dashboard.objects.get(pk=self.kwargs["pk"])
+        return kwargs
+
+    def get_success_url(self):
+        return reverse_lazy("manage-dashboard", kwargs={"pk": self.kwargs["pk"]})
+
+    def form_valid(self, form):
+        domain = form.cleaned_data["datasource"]
+        dashboard = self.get_object()
+        dashboard.whitelist_domains.add(domain)
+        messages.success(self.request, f"Successfully granted access to {domain.name}")
+
+        return FormMixin.form_valid(self, form)
```

Comment:
```suggestion
        return super().form_valid(form)
```
Can use super here also, and then the `FormMixin` import can be removed